src/presentation/ui/component/form/fileUploadTextInputFileContentReader.templ

```diff
@@ -0,0 +1,65 @@
+package componentForm
+
+script FileUploadTextInputFileContentReaderLocalState() {
+	document.addEventListener('alpine:init', () => {
+		Alpine.data('fileUploadTextInputFileContentReader', () => ({
+			uploadedFileName: '',
+			get uploadedFileNameLabel() {
+				if (this.uploadedFileName == '') {
+					return 'No file chosen';
+				}
+
+				return this.uploadedFileName;
+			},
+			init() {
+				this.uploadedFileName = '';
+			},
+			handleFileUpload(event) {
+				const inputFiles = Array.from(event.target.files);
+				if (inputFiles.length == 0) {
+					return;
+				}
+
+				const uploadedFile = inputFiles[0];
+				this.uploadedFileName = uploadedFile.name;
+
+				const reader = new FileReader();
+				reader.onload = (event) => {
+					this.$dispatch('file-content-readed', event.target.result);
+				};
+				reader.readAsText(uploadedFile);
+			}
+		}));
+	});
+}
+
+templ FileUploadTextInputFileContentReader(
+	id, label, bindValuePath, acceptAttr string,
+	denseMode bool,
+) {
+	<!-- FileUploadTextInputFileContentReader -->
+	@FileUploadTextInputFileContentReaderLocalState()
+	<div class="relative w-full" x-data="fileUploadTextInputFileContentReader">
+		<input
+			x-ref={ id + "fileInput" }
+			type="file"
+			accept={ acceptAttr }
+			class="hidden"
+			@change="handleFileUpload"
+			@file-content-readed={ bindValuePath + " = $event.detail" }
+		/>
+		<div
+			@click={ "$refs." + id + "fileInput.click()" }
+			if denseMode {
+				class="bg-os-300 border-os-200 hover:border-os-100 autofill:bg-os-300 focus:border-os-50 h-7.5 peer relative w-full cursor-default cursor-pointer rounded-md border px-1.5 pt-2 text-[13px] text-slate-400 placeholder-transparent outline-none transition-all"
+			} else {
+				class="bg-os-300 border-os-200 hover:border-os-100 autofill:bg-os-300 focus:border-os-50 peer relative h-10 w-full cursor-default cursor-pointer rounded-md border px-3 pt-2 text-sm text-slate-400 placeholder-transparent outline-none transition-all"
+			}
```

Comment:
Are you using denseMode? If not, please remove.

Reply:
Done!